source/MaterialXView/Material.cpp

```diff
@@ -401,17 +417,20 @@ bool Material::bindImage(const mx::FilePath& filename, const std::string& unifor
     if (!imageHandler->acquireImage(resolvedFilename, desc, true, fallbackColor) && !filename.isEmpty())
     {
         std::cerr << "Failed to load image: " << resolvedFilename.asString() << std::endl;
+        return returnPath;
     }
 
     // Bind the image and set its sampling properties.
-    int textureLocation = imageHandler->getBoundTextureLocation(desc.resourceId);
-    if (textureLocation < 0)
-        return false;
-
-    _glShader->setUniform(uniformName, textureLocation, false);
-    imageHandler->bindImage(resolvedFilename, samplingProperties);
-
-    return true;
+    if (imageHandler->bindImage(resolvedFilename, samplingProperties))
+    {
+        int textureLocation = imageHandler->getBoundTextureLocation(desc.resourceId);
+        if (textureLocation >= 0)
+        {
+            _glShader->setUniform(uniformName, textureLocation, false);
+            return resolvedFilename;
```

Comment:
We return the file identifier as that's what we use to find cached images and from there the resource id into the bound texture unit list.